OAuth/ResourceOwner/GenericOAuth2ResourceOwner.php

```diff
@@ -213,9 +214,17 @@ protected function configureOptions(OptionsResolverInterface $resolver)
             'use_bearer_authorization' => true,
         ));
 
-        $resolver->setOptional(array(
-            'revoke_token_url',
-        ));
+        $reflection = new ReflectionClass($resolver);
```

Comment:
Please not use `ReflectionClass`, use `method_exists($resolver, 'setDefined')` instead.

Can you refer which deprecated SF method you are fixed and for which version?